include/boost/geometry/algorithms/detail/is_valid/polyhedral_surface.hpp

```diff
@@ -0,0 +1,453 @@
+        if (intersects)
+        {
+            visitor.template apply<failure_invalid_intersection>();
+            return intersection_type::invalid;
+        }
+        return intersection_type::none;
+    }
+
+    // Helper function to check if any segment of a ring intersects a polygon
+    template <typename VisitPolicy, typename Strategy>
+    bool check_polygon_polygon_intersection(polygon_3d const& polygon1,
+                                            polygon_3d const& polygon2,
+                                            size_t const& index2,
+                                            VisitPolicy& visitor,
+                                            bool& disconnected,
+                                            Strategy const& strategy)
+    {
+        auto const& ring1 = boost::geometry::exterior_ring(polygon1);
+        for (auto it = boost::begin(ring1); it != boost::end(ring1) - 1; ++it)
+        {
+            auto result = check_polygon_segment_intersection(*it, *(it + 1), polygon2, index2,
+                                                             visitor, strategy);
+            if (result == intersection_type::invalid)
+            {
+                return false; // Found invalid intersection
+            }
+            else if (result == intersection_type::valid)
+            {
+                disconnected = false; // Found valid intersection
+            }
+        }
+        return true; // Either no intersection or valid intersection
+    }
+
+ public:
+
+    template <typename VisitPolicy, typename Strategy>
+    intersection_type apply(polygon_3d const& polygon1,
+                            polygon_3d const& polygon2,
+                            size_t const& index1,
+                            size_t const& index2,
+                            VisitPolicy& visitor,
+                            Strategy const& strategy)
+    {
+        // Check if any segment of polygon1 intersects polygon2
+        bool disconnected = true;
+        if (!check_polygon_polygon_intersection(polygon1, polygon2, index2, visitor, disconnected, strategy) ||
+            !check_polygon_polygon_intersection(polygon2, polygon1, index1, visitor, disconnected, strategy))
+        {
+            return intersection_type::invalid;
+        }
+        if (disconnected)
+        {
+            return intersection_type::none;
+        }
+        return intersection_type::valid;
+    }
+
+private:
+    std::vector<polygon_projection> m_polygon_projections;
+};
+
+
+template <typename PolyhedralSurface>
+class is_valid_polyhedral_surface
+{
+public:
+    template <typename VisitPolicy, typename Strategy>
+    static inline bool apply(PolyhedralSurface const& surface,
+                             VisitPolicy& visitor,
+                             Strategy const& strategy)
+    {
+        boost::ignore_unused(strategy);
+        bool disconnected = true;
```

Comment:
Suggestion, call it `connected = false` and set it later to `true`. For me that would read a bit better

Reply:
This is also removed. There is a similar structure/logic in `check_polygon_polygon_intersection` where I used your naming approach.